hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -555,38 +567,9 @@ public void onCompactionCompleted(RocksDB db,
               .append('\n');
         }
 
-        // Mark the beginning of a compaction log
-        sb.append(COMPACTION_LOG_ENTRY_LINE_PREFIX);
-        sb.append(db.getLatestSequenceNumber());
-        sb.append(SPACE_DELIMITER);
-
-        // Trim DB path, only keep the SST file name
-        final int filenameOffset =
-            compactionJobInfo.inputFiles().get(0).lastIndexOf("/") + 1;
-
-        // Append the list of input files
-        final List<String> inputFiles = compactionJobInfo.inputFiles();
-        // Trim the file path, leave only the SST file name without extension
-        inputFiles.replaceAll(s -> s.substring(
-            filenameOffset, s.length() - SST_FILE_EXTENSION_LENGTH));
-        final String inputFilesJoined =
-            String.join(COMPACTION_LOG_ENTRY_FILE_DELIMITER, inputFiles);
-        sb.append(inputFilesJoined);
-
-        // Insert delimiter between input files and output files
-        sb.append(COMPACTION_LOG_ENTRY_INPUT_OUTPUT_FILES_DELIMITER);
-
-        // Append the list of output files
-        final List<String> outputFiles = compactionJobInfo.outputFiles();
-        outputFiles.replaceAll(s -> s.substring(
-            filenameOffset, s.length() - SST_FILE_EXTENSION_LENGTH));
-        final String outputFilesJoined =
-            String.join(COMPACTION_LOG_ENTRY_FILE_DELIMITER, outputFiles);
-        sb.append(outputFilesJoined);
-
-        // End of line
+        sb.append(COMPACTION_LOG_ENTRY_LINE_PREFIX_V1);
+        sb.append(compactionLogEntry.toEncodedString());
         sb.append('\n');
```

Comment:
Interesting. I'm not sure `\n` would still be a valid separator now that we encode log entry with protobuf..

Ideally, the whole file would just be a `repeated CompactionLogEntryProto`. Though not sure how to append in chunks in this case.

Or, start a new RocksDB ColumnFamily that just have encoded `CompactionLogEntryProto` as its value?

Reply:
It is a good idea to use `ColumnFamily` and will address [HDDS-7601](https://issues.apache.org/jira/browse/HDDS-7601) as well. Let me think about it and see if it will cause any issue in bootstrapping. If using `ColumnFamily` doesn't cause any problem, I'll make the changes according. Otherwise will continue using Compaction log file.

Reply:
Also I think we should avoid writing to the same files/dirs as the old compaction logs' because slightly older Ozone versions won't be able to recognize the new header anyway. (e.g. upgrade OM without finalizing, then downgrade)

Reply:
Good point about upgrade and downgrade.

Reply:
You can either migrate old compaction log to the new layout (and delete old logs after migration), or just wipe out old logs on upgrade. So when downgraded it won't have any compaction log entries.